backend/globaleaks/jobs/log_sched.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+#
+#   log_sched
+#   *********
+#
+import os
+from twisted.internet.defer import inlineCallbacks, returnValue
+from storm.expr import Desc
+
+from globaleaks.jobs.base import GLJob
+from globaleaks.settings import transact, transact_ro
+from globaleaks.models import Log
+from globaleaks.utils.utility import log
+from globaleaks.utils.logger import LogQueue, LoggedEvent, initialize_LoggedEvent
+
+
+class LogSchedule(GLJob):
+    """
+    This class just flush the logs in the Database, is executed quite
+    frequently, and the goal is to record the event safely. the events
+    are kept in memory, by the globaleaks.utils.logger.Log* classes
+
+    """
+    name = "Log"
+
+    highest_logged_id = 0
+
+    @transact_ro
+    def initialize_highest_id(self, store):
+
+        max_id_check = store.find(Log)
+        max_id_check.order_by(Desc(Log.id))
+
+        if max_id_check.count() == 0:
+            log.debug("First initialization spotted!")
+            LogSchedule.highest_logged_id = 1
+        else:
+            LogSchedule.highest_logged_id = max_id_check[0].id
+
+
+    @transact
+    def dump_fresh_logs(self, store):
+        """
+        Look in the current memory storage of Logs, if there is
+        something not yet logged in the DB.
+
+        In the meanwhile, if the number of entries overcome the
+        50 elements, delete the old one. This mean that, worst case
+        scenario, we have in memory 50 Object x (Tip Numb + Receiver Num + Admin)
```

Comment:
this thing of dropping logs is wrong; for accountability reasons we cant never delete logs so that if we get the memory limits we have simply to write them on the disk and a single query of write won't weight more of a submsission so it's ok to do it automatically.

what i would suggest is:
- write when logs > memory_logs_limit
- have a scheduler that flush every minute anyhow (a single write won't weight more than all the queries performed during a submission)
- since now for reliabiliy reasons we need to handle shutdown signals (e.g.: CTRL + C)and perform flush of the data that is in memory before shutting down the app.

\cc @fpietrosanti